crates/sui-protocol-config/src/lib.rs

```diff
@@ -148,6 +148,11 @@ struct FeatureFlags {
     // Disallow adding `key` ability to types during package upgrades.
     #[serde(skip_serializing_if = "is_false")]
     disallow_adding_key_ability: bool,
+
+    // advance to highest supported protocol version at epoch change, instead of the next consecutive
+    // protocol version.
+    #[serde(skip_serializing_if = "is_false")]
+    advance_to_hightest_supported_protocol_version: bool,
```

Comment:
I'm not 100% convinced that we need this feature flag -- for a second I thought it was required for state sync but I convinced myself otherwise, so the only protection I can imagine it provides us against a situation where one node proposes upgrading to N + 1 and another node proposes upgrading to N + k.  But we can find ourselves in the same situation after this flag gets enabled if two binaries support different max versions, so is it worth gating?

Reply:
The feature flag affects the vote counting, it doesn't affect what nodes propose. So if the votes are counted differently on different nodes, we may not be able to certify the final checkpoint. So its definitely needed!